### ISSUE TITLE
Quest search filter: Uses The Horror Expansion

### ISSUE THREAD

**Issue by JTimNolan** [ExpeditionRPG/expedition#403]
Yesterday I was looking for a quest that specificially does use the horror expansion, would be nice to be able to search for it. We'd probably want to set this up in such a way that it can be used for other expansions in the future.

**Comment by toddmedema** [comment]
Agreed! Picturing a checkbox or multiselect UI that transmits a `requires[]` array as part of search params, which then gets translated into the appropriate SQL query in the /search API handler

**Comment by toddmedema** [comment]
A good candidate contribution for anyone looking to learn more about Node / SQL / the back-end

**Comment by JTimNolan** [comment]
One thing we should be thinking about for this is making sure that we can easily repeat the process / UI for future expansions. In particular, I'm wondering if we can nicely allow for differentiating a search query between e.g. "The Horror OR Expansion 2" vs "The Horror AND Expansion 2". That may fall into the nice-to-have category though

**Comment by toddmedema** [comment]
Good idea :) Thinking this might make the most sense as an OR search for now, ie these are all of the expansions I own, so show me quests that I can play. The alternative, AND'ing, runs the risk of being a too-restrictive search that frequently shows 0 results, which would be frustrating.

**Comment by nimish-gupta** [comment]
> Agreed! Picturing a checkbox or multiselect UI that transmits a `requires[]` array as part of search params, which then gets translated into the appropriate SQL query in the /search API handler

Hey @toddmedema, I am thinking of picking this one as my first issue on this repo. Can you please explain that where you want the checkbox/multiselect in UI?

**Comment by toddmedema** [comment]
Great! On the middle tab (quests), hit "Filter and Sort" in the top right

Right now, we show "For 1 player with <expansion> (based on settings)" on the top of that page. We can remove the "With <expansions>" part and a list of checkmarks for expansions (right now just The Horror and The Future).

The checkboxes should default to their matching setting state, ie if settings.contentSets.future is true, that checkbox should start enabled.

If the content set is false, that checkbox should be disabled (aka greyed out and unselectable) with a note that that content set must be enabled in settings.

Let me know if that makes sense or you have any questions!

**Comment by toddmedema** [comment]
auto-watch isn't set up for the server, but you're welcome to set it up if it's slowing you down significantly!

Note that it's currently deployed by running the start command, which must build in under 60 seconds on a Heroku instance, so we need to preserve the existing command + webpack file for `npm run start`, and should create an entirely separate command + webpack file for development mode (if you decide to add that functionality)
